src/publish.js

```diff
@@ -440,56 +453,71 @@ have published it.\
         return error;
       }
 
-      if ((version?.length > 0) || (rename?.length > 0)) {
-        let originalName;
-        if (version?.length <= 0) { version = 'patch'; }
-        if (rename?.length > 0) { originalName = pack.name; }
+      if (version?.length === 0 && tag?.length === 0 && rename?.length === 0) {
```

Comment:
I think it would be a benefit to know this is somehow still ending up `undefined` (despite the normalization we're doing) sooner, due `undefined.length` giving an error `Uncaught TypeError: Cannot read properties of undefined (reading 'length')`.

```suggestion
      if (version.length === 0 && tag.length === 0 && rename.length === 0) {
```

Sort of theoretical, since it shouldn't be possible.

But also, I like to avoid more complicated syntax if it's not needed, for ease of reading.

Also: Given the head-scratcher this issue was to solve, correctness and surfacing error messages in as straight-forward a way as possible is kind of my top priority, rather than avoiding errors in this case. Unless we want to try/catch and print the error and exit gracefully, but... I don't think that helps anything anyway in this case.

**tl;dr: Hitting a simple error message would be better than not hitting it, in this case?**

Not a blocker, since it should all be hypothetical with all the normalizing done in this PR. But the goose-chase tracking down strange behavior makes me a tad paranoid, not gonna lie.

Reply:
Sorry for all the nitpicking, just inclined to make this code _very tidy_ if possible after what had to be thought through to get here.

Reply:
_Note: If moving away from optional chaining for the `varName.length` checks, there are three more in the diff for this PR, I just didn't individually comment them all._

Reply:
P.S. I suppose these instances of optional chaining mirror what was there before, but while we're addressing tiny subtleties of behavior that were unintuitive, I think it's worth deciding whether to move away from the optional chaining or not.